### ISSUE TITLE
bug: Playback doesn't go to next video in a playlist with a SponsorBlock segment at the end of a video

### ISSUE THREAD

**Issue by bigyihsuan** [ReVanced/revanced-patches#2751]
### Bug description

When SponsorBlock skips segments that go up to the end of a video, when that video is in a playlist (e.g. Watch Later), playback fails to go to the next video and gets stuck on the end of the current video.

- Revanced on Youtube 19.04.37
- Non-root
- Vanced MicroG v0.3.0.234914

### Acknowledgements

- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.

**Comment by oSumAtrIX** [comment]
I added a video to a watch later, and SponsorBlock skipped it; the next video played. Reproduction steps are insufficient.

**Comment by bigyihsuan** [comment]
> I added a video to a watch later, and SponsorBlock skipped it; the next video played. Reproduction steps are insufficient.

What video did you add to your Watch Later? I want to see if the bug is video-specific.

**Comment by bigyihsuan** [comment]
Here is a video of this bug in action.

https://www.youtube.com/watch?v=gtwdqfTSiKQ

**Comment by oSumAtrIX** [comment]
Add the video where it occurs to the PR description. https://youtu.be/J6W1jYoa1HM

**Comment by bigyihsuan** [comment]
I did some initial half-half testing and couldn't replicate the issue. Maybe it's due to _3_ different patches interacting that's causing this.

Either way, I'll post my patch list and settings here to see if any of these seems like it's causing the issue:

### Patches

```json
{
  "com.google.android.youtube": [
    "Alternative thumbnails",
    "Always repeat",
    "Announcements",
    "Bypass URL redirects",
    "Change header",
    "Change start page",
    "Client spoof",
    "Comments",
    "Copy video URL",
    "Custom branding",
    "Custom player overlay opacity",
    "Disable auto captions",
    "Disable fullscreen ambient mode",
    "Disable player popup panels",
    "Disable precise seeking gesture",
    "Disable resuming Shorts on startup",
    "Disable rolling number animations",
    "Disable suggested video end screen",
    "Disable zoom haptics",
    "Enable debugging",
    "Enable tablet layout",
    "External downloads",
    "GmsCore support",
    "HDR auto brightness",
    "Hide Load more button",
    "Hide Shorts components",
    "Hide ads",
    "Hide album cards",
    "Hide autoplay button",
    "Hide breaking news shelf",
    "Hide captions button",
    "Hide cast button",
    "Hide crowdfunding box",
    "Hide endscreen cards",
    "Hide filter bar",
    "Hide floating microphone button",
    "Hide info cards",
    "Hide layout components",
    "Hide player buttons",
    "Hide seekbar",
    "Hide timestamp",
    "Hide video action buttons",
    "Minimized playback",
    "Navigation buttons",
    "Open links externally",
    "Playback speed",
    "Player flyout menu",
    "Remember video quality",
    "Remove player controls background",
    "Remove tracking query parameter",
    "Remove viewer discretion dialog",
    "Restore old seekbar thumbnails",
    "Restore old video quality menu",
    "Return YouTube Dislike",
    "Seekbar tapping",
    "SponsorBlock",
    "Spoof app version",
    "Spoof device dimensions",
    "Swipe controls",
    "Tablet mini player",
    "Theme",
    "Video ads",
    "Wide searchbar"
  ]
}
```

### Settings

```json
"announcement_last_hash": "B1xIZJaDpdY2wvtv20n882ZXshnYNlB8P6IHabvH6uQ=",
"disable_precise_seeking_gesture": false,
"hdr_auto_brightness": false,
"hide_crowdfunding_box": true,
"hide_gray_separator": false,
"hide_info_cards": false,
"hide_info_cards_section": false,
"hide_shorts": true,
"hide_timed_reactions": false,
"spoof_signature_in_feed_enabled": true,
"sb_filler": "skip",
"sb_highlight": "seekbar-only",
"sb_interaction": "skip",
"sb_intro": "skip",
"sb_local_time_saved_milliseconds": 1425170,
"sb_local_time_saved_number_segments": 55,
"sb_outro": "skip",
"sb_preview": "skip",
"sb_selfpromo": "skip",
"sb_sponsor": "skip"
```

**Comment by bigyihsuan** [comment]
> Did you try excluding `Disable suggested video end screen` during patching?

I did the following:

1. Turned off "disable suggested video end screen". No effect, playback still hangs at the end of the video after skipping the end segment.
2. Restarted the app with said option off, and tried again. No effect, same as above.
4. Uninstalling the old copy, patching and installing a new copy without the patch, with above settings, and an app restart. No effect, same as above.

I've found a [Reddit thread](https://www.reddit.com/r/revancedapp/comments/1as4q2t/finally_fixed_broken_autoplay/) from a few days ago that suggests to do exactly that, but as you can see it hasn't worked in my case.

**Comment by bigyihsuan** [comment]
> Did you try repatching with only `GMS core support` and `SponsorBlock`?

This was one of the first things I did, and it has the bug.

**Comment by lica98** [comment]
same issue for me, tried repatching without `Disable suggested video end screen` but to no avail 😕

**Comment by bigyihsuan** [comment]
I believe I have a consistent method of getting the bug to _not_ trigger.

In all the videos I've posted, and when I made the bug report, I had my OnePlus Buds Pro on and connected through Bluetooth. Lately, due to the earbuds dying really quickly, I have been watching videos through the built-in speaker. Maybe this bug is due to Bluetooth? I will need to test later with both my earbuds and some Bluetooth speakers I have lying around.
